coinparam/register.go

```diff
@@ -31,6 +32,26 @@ type Params struct {
 
 	// GenesisHash is the starting block hash.
 	GenesisHash *chainhash.Hash
+	
+	// The function used to calculate the proof of work value for a block
+	PoWFunction func(b []byte) chainhash.Hash
+  
+    // The function used to calculate the difficulty of a given block
+    DiffCalcFunction func(r io.ReadSeeker, height, startheight int32, p *Params) (uint32, error)
```

Comment:
Yeah, this will work, I've just got a feeling there's a better way to do it.
But I can't actually think of one so this is fine :)